scripts/run-bundled-codegen.sh

```diff
@@ -0,0 +1,64 @@
+# Bail if errors occur
+set -e
+
+# Get the path to the script directory
+SCRIPT_DIR="$(eval dirname $0)"
+
+# Get the SHASUM of the tarball
+ZIP_FILE="${SCRIPT_DIR}/apollo.tar.gz"
+SHASUM="$(eval /usr/bin/shasum -a 256 ${ZIP_FILE})"
+SHASUM_FILE="${SCRIPT_DIR}/apollo/.shasum"
+APOLLO_DIR="${SCRIPT_DIR}"/apollo
+
+# Helper functions
+remove_existing_apollo() {
+  rm -r "${APOLLO_DIR}"
+}
+
+extract_cli() {
+  tar xvzf "${SCRIPT_DIR}"/apollo.tar.gz -C "${SCRIPT_DIR}"
+  touch "${SHASUM_FILE}"
+  tee "${SHASUM_FILE}" <<<"${SHASUM}"
```

Comment:
I'm pretty sure that `<<<` is not POSIX and that it might be preferable to:

```suggestion
  echo "${SHASUM}" | tee "${SHASUM_FILE}"
```

This would accomplish the same thing and also work in `/bin/sh`, `dash`, `zsh`, etc.

Also, if printing the actual SHA in the debug output is also not desired, this could just be:

```suggestion
  echo "${SHASUM}" > "${SHASUM_FILE}"
```

This would create it and write the SHA silently. 🗣 🚫

Reply:
Btw; these scripts should only be run from a Xcode build phase, so I think we can be pretty sure which shell it will run in.